contracts/artifacts/DelegatedOptIn.approval.teal

```diff
@@ -0,0 +1,530 @@
+#pragma version 8
+	b main
+
+getSenderReceiverHash:
+	proto 2 1
+
+	// ./contracts/delegated_optin_app.algo.ts:36
+	// return sha256(concat(sender, receiverOrSigner));
+	frame_dig -1 // sender: address
+	frame_dig -2 // receiverOrSigner: address
+	concat
+	sha256
+	retsub
+
+bare_route_create:
+	txn OnCompletion
+	int NoOp
+	==
+	txn ApplicationID
+	int 0
+	==
+	&&
+	assert
+
+	// no dupn needed
+	callsub create
+	int 1
+	return
+
+create:
+	proto 0 0
+
+	// ./contracts/delegated_optin_app.algo.ts:41
+	// this.assetMBR.set(100_000)
+	byte "assetMBR"
+	int 100_000
+	app_global_put
+	retsub
+
+abi_route_setSigVerificationAddress:
+	txn OnCompletion
+	int NoOp
+	==
+	txn ApplicationID
+	int 0
+	!=
+	&&
+	assert
+
+	// no dupn needed
+	txna ApplicationArgs 1
+	callsub setSigVerificationAddress
+	int 1
+	return
+
+setSigVerificationAddress:
+	proto 1 0
+
+	// ./contracts/delegated_optin_app.algo.ts:53
+	// assert(this.txn.sender === this.app.creator)
+	txn Sender
+	txna Applications 0
+	app_params_get AppCreator
+	assert
+	==
+	assert
+
+	// ./contracts/delegated_optin_app.algo.ts:54
+	// assert(!this.sigVerificationAddress.exists())
+	txna Applications 0
+	byte "sigVerificationAddress"
+	app_global_get_ex
+	swap
+	pop
+	!
+	assert
+
+	// ./contracts/delegated_optin_app.algo.ts:55
+	// this.sigVerificationAddress.set(lsig)
+	byte "sigVerificationAddress"
+	frame_dig -1 // lsig: address
+	app_global_put
+	retsub
+
+abi_route_updateAssetMBR:
+	txn OnCompletion
+	int NoOp
+	==
+	txn ApplicationID
+	int 0
+	!=
+	&&
+	assert
+	byte 0x; dup
+	txna ApplicationArgs 1
+	btoi
+	txnas Assets
+	callsub updateAssetMBR
+	int 1
+	return
+
+updateAssetMBR:
```

Comment:
Yes that would be very useful. Exposing as much of the parameter tables as possible in general would be great.

Reply:
I suppose I've avoided it in the hopes that programs would always use the pattern "Get MBR, do something, compute delta". But here you're trying to determine what a transaction would do even though that transaction already happened, at the top-level. You don't have a chance to get the mbr before and after. (So that's what you do, artificially, in this method.)

Reply:
Yeah I think if there were to be an opcode like that it would only be useful in a scenario like this. Not sure if it'd be worth waiting for the next AVM version just for this functionality.